curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/Reaper.java

```diff
@@ -333,7 +355,20 @@ public void notLeader()
                 reapingIsActive.set(false);
             }
         };
-        localLeaderLatch.addListener(listener);
-        return localLeaderLatch;
+        leaderLatch.addListener(listener);
+
+        reapingIsActive.set(leaderLatch.hasLeadership());
+    }
+
+    private static LeaderLatch makeLeaderLatchIfPathNotNull(CuratorFramework client, String leaderPath)
```

Comment:
It's because it's called from within a constructor that calls another constructor.